cmd/releasego/update-azure-linux.go

```diff
@@ -286,15 +351,38 @@ func updateCGManifest(buildAssets *buildassets.BuildAssets, cgManifestContent []
 	}
 
 	updated := false
-	for i := range cgManifest.Registrations {
-		registration := &cgManifest.Registrations[i].Component.Other
-		if registration.Name == "golang" {
-			registration.Version = buildAssets.GoVersion().MajorMinorPatch()
-			registration.DownloadURL = fmt.Sprintf(
-				"https://github.com/microsoft/go/releases/download/%s/%s",
+	for i, reg := range cgManifest.Registrations {
+		registration := make(map[string]interface{})
+		if err := json.Unmarshal(reg, &registration); err != nil {
+			return nil, fmt.Errorf("failed to parse registration in cgmanifest.json: %w", err)
+		}
+
+		component, ok := registration["component"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		other, ok := component["other"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		if name, ok := other["name"].(string); ok && name == "golang" {
+			other["version"] = buildAssets.GoVersion().MajorMinorPatch()
+			other["downloadUrl"] = fmt.Sprintf(
+				"https://github.com/microsoft/go/releases/download/v%s/%s",
 				buildAssets.GoVersion().Full(),
 				path.Base(buildAssets.GoSrcURL),
 			)
+
+			buf := new(bytes.Buffer)
+			encoder := json.NewEncoder(buf)
+			encoder.SetEscapeHTML(false)
+			if err := encoder.Encode(registration); err != nil {
```

Comment:
The idea of this general area is to preserve the order of the original keys, right? (Which is something I think we do need to do--to avoid thrashing with other bots (or opinionated humans).)

I think since `registration` is a map, this line ends up sorting it again anyway.

Maybe we could implement something like https://github.com/golang/go/issues/27179#issuecomment-587528269 to preserve order without quite as much complication?

It could also be simpler to treat the file as only text rather than actually fully parsing it as JSON, so it's easier to leave most of the file untouched no matter what other people do to the file.

Reply:
> The idea of this general area is to preserve the order of the original keys, right? (Which is something I think we do need to do--to avoid thrashing with other bots (or opinionated humans).)
> 
> I think since `registration` is a map, this line ends up sorting it again anyway.
> 
> Maybe we could implement something like [golang/go#27179 (comment)](https://github.com/golang/go/issues/27179#issuecomment-587528269) to preserve order without quite as much complication?
> 
> It could also be simpler to treat the file as only text rather than actually fully parsing it as JSON, so it's easier to leave most of the file untouched no matter what other people do to the file.

So I did the another thing mentioned in same issue:
https://github.com/golang/go/issues/27179#issuecomment-848662008

Reply:
My worry here is that a bot will change the key order back to what it was, no matter how minimal the difference is. Then next time we run the update, our bot will change it back to "our" order, and then they'll swap it back, and so on. (And, not necessarily an Azure Linux bot: any other team doing auto-updates too.)

Ultimately this just means things are harder to review/`blame` and the Git repo might get a tiny bit larger, so I suppose it's up to Azure Linux. And, if they really want to avoid this problem, they need to add CI that verifies that this file is laid out in a prescriptive order.

Reply:
Fixed with using strudt instead of map for json marshalling.